internal/hostname/httputils/helpers.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+// This file is pulled from datadog-agent/pkg/util/http (Only removing agent SSL config and unused funcs)
+
+package httputils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"time"
+)
+
+func createTransport() *http.Transport {
+	return &http.Transport{
```

Comment:
We should handle the case where the user has a proxy, looks like the default tracer client [does it](https://github.com/DataDog/dd-trace-go/blob/1abcec80856ed4015f240c69b46cf256ff250969/ddtrace/tracer/transport.go#L46).

Reply:
nice catch here 👍